lib/modules/manager/azure-pipelines/extract.ts

```diff
@@ -23,33 +27,43 @@ const AzurePipelinesTaskRegex = regEx(/^(?<name>[^@]+)@(?<version>.*)$/);
 
 export function extractRepository(
   repository: Repository,
+  currentRepository?: string,
 ): PackageDependency | null {
   let repositoryUrl = null;
 
+  let depName = repository.name;
+
   if (repository.type === 'github') {
     repositoryUrl = `https://github.com/${repository.name}.git`;
   } else if (repository.type === 'git') {
-    // "git" type indicates an AzureDevOps repository.
-    // The repository URL is only deducible if we are running on AzureDevOps (so can use the endpoint)
-    // and the name is of the form `Project/Repository`.
-    // The name could just be the repository name, in which case AzureDevOps defaults to the
-    // same project, which is not currently accessible here. It could be deduced later by exposing
-    // the repository URL to managers.
-    // https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/resources-repositories-repository?view=azure-pipelines#types
     const platform = GlobalConfig.get('platform');
     const endpoint = GlobalConfig.get('endpoint');
+
     if (platform === 'azure' && endpoint) {
+      // extract the project name if the repository from which the pipline is referencing templates contains the Azure DevOps project name
       if (repository.name.includes('/')) {
         const [projectName, repoName] = repository.name.split('/');
+        depName = `${projectName}/${repository.name}`;
         repositoryUrl = joinUrlParts(
           endpoint,
           encodeURIComponent(projectName),
           '_git',
           encodeURIComponent(repoName),
         );
+
+        // if the repository from which the pipline is referencing templates does not contain the Azure DevOps project name, get the project name from the repository containing the pipeline file being process
+      } else if (currentRepository && currentRepository.includes('/')) {
+        const projectName = currentRepository.split('/')[0];
+        depName = `${projectName}/${repository.name}`;
+        repositoryUrl = joinUrlParts(
+          endpoint,
+          encodeURIComponent(projectName),
+          '_git',
+          encodeURIComponent(repository.name),
+        );
       } else {
         logger.debug(
-          'Renovate cannot update repositories that do not include the project name',
+          "Renovate cannot update Azure pipelines in 'git' repositories when neither the current repository nor the target repository contains the Azure DevOps project name.",
```

Comment:
use single quotes and remove quotes from git

Reply:
Updated.